Automattic/jetpack
# Contact Form: Allow users to add multiple options to a dropdown field

Fixes https://github.com/Automattic/jetpack/issues/37565

## Proposed changes:

A recent Gutenberg update (from 18.3 to [18.4.0 RC1](https://github.com/WordPress/gutenberg/releases/tag/v18.4.0-rc.1)) broke the Contat Form dropdown field: users could not add more than one option. The issue was caused by [the `onSplit` prop being removed from the RichText component](https://github.com/WordPress/gutenberg/pull/54543).

<img width="500" alt="Screenshot 2024-06-06 at 10 28 31 AM" src="https://github.com/Automattic/jetpack/assets/1620183/5b9a67ae-cab5-4e4e-8cf6-d87db036704f">


This PR fixes it by adding a listener to the Enter key and applying similar logic to that implemented in the `onSplit` callback.

### Other information:


## Jetpack product discussion


n/a

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






_Note: I personally tested the fix in Chrome, Safari, and Firefox._

### To reproduce the issue
- Spin up a Jetpack test site (the issue happens with self-hosted, simple, and atomic sites)
- If you're using a self-hosted site, you may need to install the [Gutenberg plugin](https://github.com/WordPress/gutenberg/releases) manually to ensure the version is 18.4 or above
- Create a new post
- Add a Contact Form block
- Add a dropdown field
- Notice you cannot add more than one option: hitting the Enter keys does nothing

### To test the fix
- Checkout this branch
- Repeat the steps above
- Verify that you can add several options
- Check that the dropdown renders as expected in the frontend
- Optionally, test the fix with other types of site
